libs/counterfactuals/src/lib/CounterfactualChartLegend.tsx

```diff
@@ -0,0 +1,152 @@
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public render(): React.ReactNode {
+    const classNames = counterfactualChartStyles();
+    return (
+      <Stack className={classNames.legendAndText}>
+        <ComboBox
+          id={"CounterfactualSelectedDatapoint"}
+          className={classNames.legendLabel}
+          label={localization.Counterfactuals.selectedDatapoint}
+          onChange={this.selectPointFromDropdown}
+          options={this.getDataOptions()}
+          selectedKey={`${this.props.selectedPointsIndexes[0]}`}
+          ariaLabel={"datapoint picker"}
+          useComboBoxAsMenuWidth
+          styles={FluentUIStyles.smallDropdownStyle}
+        />
+        <div className={classNames.legendLabel}>
+          <b>{`${this.getTargetDescription()}: `}</b>
+          {getCurrentLabel(
+            this.context.dataset.task_type,
+            this.props.data.desired_range,
+            this.props.data.desired_class
+          )}
+        </div>
+        <PrimaryButton
+          className={classNames.legendLabel}
+          onClick={this.props.togglePanel}
+          disabled={this.disableCounterfactualPanel()}
+          text={
+            this.context.requestPredictions
+              ? localization.Counterfactuals.createWhatIfCounterfactual
+              : localization.Counterfactuals.createCounterfactual
+          }
+        />
+        {this.props.customPoints.length > 0 && (
+          <InteractiveLegend
+            items={this.props.customPoints.map((row, rowIndex) => {
+              return {
+                activated: this.props.customPointIsActive[rowIndex],
+                color:
+                  FluentUIStyles.fluentUIColorPalette[
+                    rowIndex + WhatIfConstants.MAX_SELECTION + 1
+                  ],
+                index: rowIndex,
+                name: row[WhatIfConstants.namePath],
+                onClick: this.props.toggleCustomActivation,
+                onDelete: this.props.removeCustomPoint
+              };
+            })}
+          />
+        )}
+      </Stack>
+    );
+  }
+
+  private getTargetDescription(): string {
+    if (this.context.dataset.task_type === "regression") {
+      return localization.Counterfactuals.currentRange;
+    }
+    return localization.Counterfactuals.currentClass;
+  }
+
+  private selectPointFromDropdown = (
+    _event: React.FormEvent<IComboBox>,
+    item?: IComboBoxOption
+  ): void => {
+    if (typeof item?.key === "string") {
+      const index = Number.parseInt(item.key);
+      this.props.setTemporaryPointToCopyOfDatasetPoint(index);
+      this.props.toggleSelectionOfPoint(index);
+      this.logTelemetryEvent(
+        TelemetryEventName.CounterfactualNewDatapointSelectedFromDropdown
+      );
+    }
+  };
+
+  private disableCounterfactualPanel = (): boolean => {
+    return (
+      this.props.selectedPointsIndexes[0] === undefined ||
```

Comment:
I see comparison to undefined in several places, but I recall Ke telling me we should not compare to null or undefined.  Perhaps it's fine to keep it as is, especially since this is just moving code around, but have you heard of this general rule?

Reply:
Yeah we have this @typescript-eslint/no-null-keyword rule: https://palantir.github.io/tslint/rules/no-null-keyword/
This rule asks us to only use undefined, instead of null

`JavaScript originally intended undefined to refer to a value that doesn’t yet exist, while null was meant to refer to a value that does exist but points to nothing. To remove confusion over the two similar values, it’s better to stick with just undefined.`